src/components/app-top-bar.js

```diff
@@ -232,12 +224,12 @@ const AppTopBar = ({ user, tabIndex, onChangeTab, userManager }) => {
         (optionInfos) => {
             onChangeTab(STUDY_VIEWS.indexOf(StudyView.MAP)); // switch to map view
             if (optionInfos.type === EQUIPMENT_TYPE.SUBSTATION.name) {
```

Comment:
openDiagramView(optionInfos.id, optionInfos.type); ?
So you can remove the "if"

Reply:
`EQUIPMENT_TYPE.SUBSTATION.name` and `SvgType.SUBSTATION` could be different.
This code is not mine, I don't think I should modify it in this refactoring.